components/ui/autocomplete/src/test/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditTextTest.kt

```diff
@@ -424,4 +433,89 @@ class InlineAutocompleteEditTextTest {
         // assigning here so it verifies the setter, not the getter
         verify(et, never()).isEnabled = true
     }
+
+    @Test
+    fun `WHEN onTextContextMenuItem is called for options other than paste THEN we should not paste() and just call super`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.copy)
+        editText.onTextContextMenuItem(android.R.id.shareText)
+        editText.onTextContextMenuItem(android.R.id.cut)
+        editText.onTextContextMenuItem(android.R.id.selectAll)
+
+        verify(editText, never()).paste(anyInt(), anyInt(), anyBoolean())
+        verify(editText, times(4)).callOnTextContextMenuItemSuper(anyInt())
+    }
+
+    @Test
+    fun `WHEN onTextContextMenuItem is called for paste THEN we should paste() and not call super`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.paste)
+
+        verify(editText).paste(anyInt(), anyInt(), anyBoolean())
+        verify(editText, never()).callOnTextContextMenuItemSuper(anyInt())
+    }
+
+    @Test
+    fun `WHEN onTextContextMenuItem is called for pasteAsPlainText THEN we should paste() and not call super`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.pasteAsPlainText)
+
+        verify(editText).paste(anyInt(), anyInt(), anyBoolean())
+        verify(editText, never()).callOnTextContextMenuItemSuper(anyInt())
+    }
+
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.LOLLIPOP, Build.VERSION_CODES.LOLLIPOP_MR1])
+    fun `GIVEN an Android L device, WHEN onTextContextMenuItem is called for paste THEN we should paste() with formatting`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.paste)
+
+        verify(editText).paste(anyInt(), anyInt(), eq(true))
+    }
+
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.M, Build.VERSION_CODES.N, Build.VERSION_CODES.O, Build.VERSION_CODES.P])
+    fun `GIVEN an Android M device, WHEN onTextContextMenuItem is called for paste THEN we should paste() without formatting`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.paste)
+
+        verify(editText).paste(anyInt(), anyInt(), eq(false))
+    }
+
+    @Test
+    fun `WHEN no previous text WHEN paste is selected THEN paste() should be called with 0,0`() {
+        val editText = spy(InlineAutocompleteEditText(testContext, attributes))
+
+        editText.onTextContextMenuItem(android.R.id.paste)
+
+        verify(editText).paste(eq(0), eq(0), eq(false))
+    }
+
+    @Test
+    fun `WHEN 5 chars previous text text WHEN paste is selected THEN paste() should be called with 0,5`() {
```

Comment:
nit: text text

Reply:
Thank you. Also replaced the first `WHEN`s with `GIVEN`s.